pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/DispatchablePlanVisitor.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.planner.physical;
+
+import org.apache.pinot.query.planner.StageMetadata;
+import org.apache.pinot.query.planner.stage.AggregateNode;
+import org.apache.pinot.query.planner.stage.FilterNode;
+import org.apache.pinot.query.planner.stage.JoinNode;
+import org.apache.pinot.query.planner.stage.MailboxReceiveNode;
+import org.apache.pinot.query.planner.stage.MailboxSendNode;
+import org.apache.pinot.query.planner.stage.ProjectNode;
+import org.apache.pinot.query.planner.stage.SortNode;
+import org.apache.pinot.query.planner.stage.StageNode;
+import org.apache.pinot.query.planner.stage.StageNodeVisitor;
+import org.apache.pinot.query.planner.stage.TableScanNode;
+import org.apache.pinot.query.planner.stage.ValueNode;
+import org.apache.pinot.query.planner.stage.WindowNode;
+
+
+public class DispatchablePlanVisitor implements StageNodeVisitor<Void, DispatchablePlanContext> {
+  public static final DispatchablePlanVisitor INSTANCE = new DispatchablePlanVisitor();
+
+  private DispatchablePlanVisitor() {
+  }
+
+  /**
+   * Entry point
+   * @param globalReceiverNode
+   * @param physicalPlanContext
+   */
+  public void constructDispatchablePlan(StageNode globalReceiverNode, DispatchablePlanContext physicalPlanContext) {
+    globalReceiverNode.visit(DispatchablePlanVisitor.INSTANCE, physicalPlanContext);
+    computeWorkerAssignment(globalReceiverNode, physicalPlanContext);
```

Comment:
nit: we can remove `constructDispatchablePlan` by moving this `computeWorkerAssignment` call to `visitMailboxReceive`. The caller can then directly do `rootNode.visit(DispatchablePlanVisitor.INSTANCE, context)`:

```
// special case for the global mailbox receive node
    if (node.getStageId() == 0) {
      context.getQueryPlan().getQueryStageMap().put(0, node);
      computeWorkerAssignment(globalReceiverNode, physicalPlanContext);
    }
```

Reply:
yeah I was hoping to now refactor this part. and make the entry point single and clean